applications/system-service/systemapi.cpp

```diff
@@ -12,6 +13,14 @@ void SystemAPI::PrepareForSleep(bool suspending){
             resumeApp = nullptr;
         }
         drawSleepImage();
+        if (DeviceSettings::instance().getDeviceType() == DeviceType::RM2) {
+            // RM2 needs some time to draw sleep image
+            struct timespec args{
```

Comment:
Did you do any checking to see if we can get away with a shorter amount of time?

Reply:
So 0.5s seems fine for the default suspend image. Could you do some followup testing with more complex images that include grey to see if that still holds up? Ideally we could use MXCFB_WAIT_FOR_UPDATE_COMPLETE but rm2fb doesn't support it yet from what I understand from talking to @raisjn